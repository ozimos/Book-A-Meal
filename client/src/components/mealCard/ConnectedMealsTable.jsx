import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';

import { connect } from 'react-redux';
import { orderActions } from '../../redux/actions';


class MealsTable extends Component {

  componentDidMount() {
    this.props.dispatch(orderActions.getMealsInOrder(this.props.id));
  }

  getNextPage = (page) => {
    const { orders, id } = this.props;
    const { pagination }
     = orders.find(order => order.id === id);
    const { limit } = pagination;
    const offset = limit * page;
    this.props.dispatch(orderActions
      .getMealsInOrder({ limit, offset }));
  }
  render() {
    const columns = [
      {
        headerStyle: { height: '0' },
        id: 'pivot',
        accessor: () => '',
        width: 30
      }, {
        headerStyle: { height: '0' },
        accessor: 'title',
        width: 400
      }, {
        headerStyle: { height: '0' },
        accessor: 'MealOrders.quantity',
        width: 150
      }, {
        headerStyle: { height: '0' },
        accessor: 'price',
        width: 150
      }, {
        headerStyle: { height: '0' },
        aggregate: vals => vals.reduce((accum, val) => accum + val),
        accessor: 'subTotal',
        width: 100
      }
    ];
    const { orders, id } = this.props;
    const { Meals, pagination = {}, connecting }
     = orders.find(order => order.id === id);

    return (
      <ReactTable
        data={Meals}
        columns={columns}
        minRows={0}
        defaultPageSize={pagination.limit || 5}
        pivotBy={['pivot']}
        loading={connecting}
        pages={pagination.pages || 1}
        onPageChange={this.getNextPage}
        collapseOnSortingChange={false}
        nextText=">>"
        previousText="<<"
      />
    );
  }
}
MealsTable.defaultProps = {
  orders: [],
};
MealsTable.propTypes = {
  id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  orders: PropTypes.arrayOf(PropTypes.object),
};
const mapStateToProps = state => ({
  orders: state.orderReducer.orders,
});
export { MealsTable };
export default connect(mapStateToProps)(MealsTable);
