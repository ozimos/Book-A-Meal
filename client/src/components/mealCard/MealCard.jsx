import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import Formsy from 'formsy-react';
import Dropzone from 'react-dropzone';
import MyFormsyInput from '../helpers/MyInput';
import MyFormsyTextArea from '../helpers/MyTextArea';
import ProgressLoader from '../helpers/ProgressLoader';
import { mealActions } from '../../redux/actions';
import imageUpload from '../../services/imageUpload';

ReactModal.setAppElement(document.getElementById('root'));
class PlainMealCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      canSubmit: false,
      showModal: false,
      displayImage: this.props.imageUrl,
      uploading: false,
      uploadPercent: 0
    };
    this.handleDrop = this.handleDrop.bind(this);
    this.setUploadPercent = this.setUploadPercent.bind(this);
  }
  setUploadPercent(percentProgress) {
    this.setState({ uploadPercent: percentProgress });
  }
  handleDrop(files) {
    this.setState({ uploading: true });
    imageUpload(files, this.setUploadPercent).then((response) => {
      const { data } = response;
      const fileURL = data.secure_url;
      this.urlInput.props.setValue(fileURL);
      this.setState({ displayImage: fileURL, uploading: false, uploadPercent: 0 });

    });
  }
  handleOpenModal = () =>
    this.setState({ showModal: true });
  handleCloseModal = () =>
    this.setState({ showModal: false });
  deleteMeal = () =>
    this.props.dispatch(mealActions.deleteMeal(this.props.id));

  handleSubmit = async (meal) => {
    await this.props.dispatch(mealActions.updateMeal(meal, this.props.id));
    this.handleCloseModal();
  }
  disableButton = () =>
    this.setState({ canSubmit: false });

  enableButton = () =>
    this.setState({ canSubmit: true });
  serverFeedback = error =>
    this.formEl.updateInputsWithError(error);
  render() {
    const { title, imageUrl, price, description } = this.props;

    return (
      <React.Fragment>
        <div className="card">
          <img src={imageUrl} alt="Meal" className="fluid-img" />
          <div className="title-element flexbox wrap">
            <button className="modal-open modal-trigger" onClick={this.handleOpenModal}>
              {title}
            </button>
            <button className="btn title-button icon" onClick={this.deleteMeal}>
                &#10006;
            </button>
          </div>
        </div>
        <ReactModal
          isOpen={this.state.showModal}
          contentLabel="Input Modal"
          className="modal-content"
          onRequestClose={this.handleCloseModal}
          shouldCloseOnOverlayClick
        >
          <div className="title flexbox">
            <h4 className="shrink">
              Meal Editor
            </h4>
            <button className="btn title-button" onClick={this.handleCloseModal}>
                &#10006;
            </button>
          </div>
          <Formsy
            className="form3"
            onValidSubmit={this.handleSubmit}
            onValid={this.enableButton}
            onInvalid={this.disableButton}
            ref={(form) => { this.formEl = form; }}
          >
            <MyFormsyInput
              typeOfInput="text"
              name="title"
              placeholder={title || 'Meal Title'}
              validations="minLength:1"
              validationError="Please enter the meal title"
            />
            <MyFormsyInput
              typeOfInput="number"
              name="price"
              placeholder={price || 'Price'}
              validations={{
                  isOnlyInt: (values, value = '') => /^(\s?|[1-9]\d*)$/.test(value)
                }}
              validationError="Please enter a price"
              validationErrors={{ isOnlyInt: 'price must be integer' }}
            />
            <MyFormsyTextArea
              name="description"
              placeholder={description || 'Description'}
            />

            <MyFormsyInput
              ref={(urlInput) => { this.urlInputMain = urlInput; }}
              innerRef={(c) => { this.urlInput = c; }}
              style={{ display: 'none' }}
              typeOfInput="url"
              name="imageUrl"
                // initialValue={this.state.displayImage}
              validations="minLength:5"
              validationError="Please select an image"
            />
          </Formsy>
          <Dropzone
            onDrop={this.handleDrop}
            multiple
            accept="image/*"
            className="dropzone"
          >
            {!this.state.uploading &&
            <button className="btn"> Select an Image</button>}
          </Dropzone>
          {this.state.uploading &&
          <ProgressLoader upload={this.state.uploadPercent} />
            }
          <div id="meal_image">
            <img src={this.state.displayImage} alt="meal" className="fluid-img" />
          </div>
          <button
            className={this.state.canSubmit ? 'btn' : 'btn btn-disabled'}
            onClick={() => this.formEl.submit()}
            type="submit"
            disabled={!this.state.canSubmit}
          >
            Continue
          </button>
        </ReactModal>
      </React.Fragment>
    );
  }
}
PlainMealCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
};
export { PlainMealCard };
export default connect(state => state)(PlainMealCard);

