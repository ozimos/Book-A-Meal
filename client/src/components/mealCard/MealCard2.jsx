import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import { hot } from 'react-hot-loader';
import { mealActions } from '../../redux/actions';
import '../../../public/styles/book_a_meal.css';
import '../../../public/styles/auth.scss';
import '../../../public/styles/modalOpenButton.scss';

ReactModal.setAppElement(document.getElementById('root'));
class MealCard2 extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
    };
  }
  handleOpenModal = () =>
    this.setState({ showModal: true });
  handleCloseModal = () =>
    this.setState({ showModal: false });

  handleSubmit = async (meal) => {
    await this.props.dispatch(mealActions.updateMeal(meal, this.props.id));
    this.handleCloseModal();
  }
  render() {
    const { id, title, imageUrl, price, description } = this.props;
    const meal = { id, title, imageUrl, price, description };
    return (
      <React.Fragment>
        <div className="card">
          <img src={imageUrl} alt="Meal" className="fluid-img" />
          <div className="title-element flexbox wrap">
            <button className="modal-open shrink modal-trigger" onClick={this.handleOpenModal}>
              {title}
            </button>
            <div className="flexbox">
              {this.props.addToMenu &&
              <button className="btn title-button icon" onClick={() => this.props.addToMenu(meal)}>
                &#10004;
              </button>}
              {this.props.removeFromMenu &&
              <button className="btn title-button icon" onClick={() => this.props.removeFromMenu(id)}>
                &#10006;
              </button>}
            </div>

          </div>
        </div>
        <ReactModal
          isOpen={this.state.showModal}
          contentLabel="Input Modal"
          onRequestClose={this.handleCloseModal}
          shouldCloseOnOverlayClick
        >
          <div className="title flexbox">
            <h3 className="shrink">
              Meal Details
            </h3>
            <div className="flexbox">
              <button className="btn title-button" onClick={this.handleCloseModal}>
                &#10006;
              </button>
            </div>
          </div>
          <div className="form-box">
            <div>{title}</div>
            <div>{price}</div>
            <div>{description}</div>
            <div>{imageUrl}</div>
          </div>
        </ReactModal>
      </React.Fragment>
    );
  }
}
MealCard2.defaultProps = {
  addToMenu: null,
  removeFromMenu: null,
};
MealCard2.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
  addToMenu: PropTypes.func,
  removeFromMenu: PropTypes.func,
};
export default connect(state => state)(hot(module)(MealCard2));

