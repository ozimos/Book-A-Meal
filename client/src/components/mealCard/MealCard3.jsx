import React from 'react';
import PropTypes from 'prop-types';
import '../../../public/styles/mealcard3.scss';

const MealCard3 = ({ id, title, imageUrl, price, description, ...props }) => {
  const meal = { id, title, imageUrl, price, description };
  return (
    <React.Fragment>
      <div className="card card-booking pillar">
        <img src={imageUrl} alt="Meal" className="fluid-img" />
        <div className="pillar">
          <div className="title-element text_left long_string">
            {title}
          </div>
          <div className="flexbox">
            <div className="text_left pad">
          &#8358;{price}
            </div>
            {props.addToOrder &&
            <button className="btn title-button" onClick={() => props.addToOrder(meal)} >
                Add to Cart
            </button>}
            {props.removeFromOrder &&
            <button className="btn title-button icon" onClick={() => props.removeFromOrder(id)}>
                &#10006;
            </button>}
          </div>
        </div>
      </div>
    </React.Fragment>
  );
};

MealCard3.defaultProps = {
  addToOrder: null,
  removeFromOrder: null,
};
MealCard3.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  addToOrder: PropTypes.func,
  removeFromOrder: PropTypes.func,
};
export default MealCard3;

