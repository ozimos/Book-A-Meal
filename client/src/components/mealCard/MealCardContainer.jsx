import React from 'react';
import PropTypes from 'prop-types';
import { hot } from 'react-hot-loader';

import '../../../public/styles/book_a_meal.css';


const MealCardContainer = ({ MealCard, meals, ...props }) =>
  (
    <div className={props.addClass ? `${props.addClass} gallery` : 'gallery'}>
      {meals.map(meal => (<MealCard
        key={meal.id}
        {...meal}
        {...props}
      />))
        }
    </div>
  );
MealCardContainer.defaultProps = {
  addClass: PropTypes.string
};
MealCardContainer.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  MealCard: PropTypes.func.isRequired,
  addClass: PropTypes.string,
};

export default hot(module)(MealCardContainer);

