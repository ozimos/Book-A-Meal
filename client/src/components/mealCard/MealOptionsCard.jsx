import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import { connect } from 'react-redux';
import Formsy from 'formsy-react';
import Dropzone from 'react-dropzone';
import MyFormsyInput from '../helpers/MyInput';
import MyFormsyTextArea from '../helpers/MyTextArea';
import ProgressLoader from '../helpers/ProgressLoader';
import { mealActions } from '../../redux/actions';
import imageUpload from '../../services/imageUpload';

ReactModal.setAppElement(document.getElementById('root'));
class BaseMealOptionsCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      canSubmit: false,
      showMealEditorModal: false,
      showDeleteConfirmModal: false,
      displayImage: this.props.imageUrl,
      uploading: false,
      uploadPercent: 0
    };
    this.handleDrop = this.handleDrop.bind(this);
    this.setUploadPercent = this.setUploadPercent.bind(this);
  }
  setUploadPercent(percentProgress) {
    this.setState({ uploadPercent: percentProgress });
  }
  handleDrop(files) {
    this.setState({ uploading: true });
    imageUpload(files, this.setUploadPercent).then((response) => {
      const { data } = response;
      const fileURL = data.secure_url;
      this.urlInput.props.setValue(fileURL);
      this.setState({
        displayImage: fileURL,
        uploading: false,
        uploadPercent: 0
      });

    });
  }
  openMealEditorModal = () =>
    this.setState({ showMealEditorModal: true });
  closeMealEditorModal = () =>
    this.setState({ showMealEditorModal: false });
    openDeleteConfirmModal = () =>
      this.setState({ showDeleteConfirmModal: true });
  closeDeleteConfirmModal = () =>
    this.setState({ showDeleteConfirmModal: false });
  deleteMeal = () =>
    this.props.dispatch(mealActions.deleteMeal(this.props.id));

  handleSubmit = async (meal) => {
    await this.props.dispatch(mealActions.updateMeal(meal, this.props.id));
    this.closeMealEditorModal();
  }
  disableButton = () =>
    this.setState({ canSubmit: false });

  enableButton = () =>
    this.setState({ canSubmit: true });
  serverFeedback = error =>
    this.formEl.updateInputsWithError(error);
  render() {
    const { title, imageUrl, price, description } = this.props;

    return (
      <React.Fragment>
        <div className="card">
          <div className="overlay-container">
            <div className="overlay" >
              <div className="flexbox">
                <span className="long_string text_left">
                  {title}
                </span>
                <span>
                &#8358;{price}
                </span>
              </div>
            </div>
            <img src={imageUrl} alt="Meal" className="fluid-img" />
          </div>
          <p className="descrip long_string">
            {description}
          </p>
          <button
            className="btn pad-btn modal-open"
            onClick={this.openMealEditorModal}
          >
              Edit
          </button>
          <button
            className="title-button card-btn"
            onClick={this.openDeleteConfirmModal}
          >
                &#10006;
          </button>
        </div>
        <ReactModal
          isOpen={this.state.showMealEditorModal}
          contentLabel="Input Modal"
          className="modal-content"
          onRequestClose={this.closeMealEditorModal}
          shouldCloseOnOverlayClick
        >
          <div className="title flexbox navbar-fixed">
            <h4 className="shrink">
              Meal Editor
            </h4>
            <button
              className="btn title-button"
              onClick={this.closeMealEditorModal}
            >
                &#10006;
            </button>
          </div>
          <div className="form3-grid">
            <div>
              <Formsy
                className="form3"
                onValidSubmit={this.handleSubmit}
                onValid={this.enableButton}
                onInvalid={this.disableButton}
                ref={(form) => { this.formEl = form; }}
              >
                <MyFormsyInput
                  typeOfInput="text"
                  name="title"
                  placeholder={title || 'Meal Title'}
                  validations={{ minLength: 1, maxLength: 48 }}
                  validationError="Please enter the meal title"
                  validationErrors={{
                minLength: 'input must be longer than 1 character',
                maxLength: 'input must be shorter than 50 characters',
              }}
                />
                <MyFormsyInput
                  typeOfInput="number"
                  name="price"
                  placeholder={`\u20A6${price}` || 'Price'}
                  validations={{
                  isOnlyInt: (values, value = '') =>
                  /^(\s?|[1-9]\d*)$/.test(value),
                  minLength: 1,
                  maxLength: 48
                }}
                  validationError="Please enter a price"
                  validationErrors={{
                isOnlyInt: 'price must be integer',
                minLength: 'input must be longer than 1 character',
                maxLength: 'input must be shorter than 50 characters',
              }}
                />
                <MyFormsyTextArea
                  name="description"
                  placeholder={description || 'Description'}
                  validations={{ minLength: 5, maxLength: 98 }}
                  validationErrors={{
                minLength: 'input must be longer than 5 characters',
                maxLength: 'input must be shorter than 100 characters',
              }}
                />

                <MyFormsyInput
                  ref={(urlInput) => { this.urlInputMain = urlInput; }}
                  innerRef={(c) => { this.urlInput = c; }}
                  style={{ display: 'none' }}
                  typeOfInput="url"
                  name="imageUrl"
                  validations={{ isUrl: true, minLength: 5, maxLength: 98 }}
                  validationError="Please select an image"
                  validationErrors={{
                isUrl: 'A valid url was not supplied',
                minLength: 'input must be longer than 5 characters',
                maxLength: 'input must be shorter than 100 characters',
              }}
                />
              </Formsy>
            </div>
            <div className="overlay-container">

              <div className="overlay full" >
                <Dropzone
                  onDrop={this.handleDrop}
                  multiple
                  accept="image/*"
                  className="dropzone"
                >
                  {!this.state.uploading &&
                  <button className="btn"> Select an Image</button>}

                </Dropzone>

              </div>
              <div id="meal_image">
                {this.state.uploading &&
                <ProgressLoader upload={this.state.uploadPercent} />
                }
                {!this.state.uploading &&
                <img
                  src={this.state.displayImage}
                  alt="meal"
                  className="fluid-img"
                />}
              </div>
            </div>
            <button
              className={(this.state.canSubmit || !this.props.connecting)
              ? 'btn' : 'btn btn-disabled'}
              onClick={() => this.formEl.submit()}
              type="submit"
              disabled={!this.state.canSubmit || this.props.connecting}
            >
              <p>Continue</p>
            </button>
          </div>


        </ReactModal>
        <ReactModal
          isOpen={this.state.showDeleteConfirmModal}
          contentLabel="Confirmation Modal"
          className="modal-content"
          onRequestClose={this.closeMealEditorModal}
          shouldCloseOnOverlayClick={false}
        >
          <p>
             This action will delete your meal. Do you want to continue
          </p>
          <div className="title flexbox">
            <button
              style={{ backgroundColor: 'red !important' }}
              className="btn"
              onClick={() => {
                this.deleteMeal();
                this.closeDeleteConfirmModal();
              }}
            >
              <p>Yes</p>
            </button>
            <button
              className="btn"
              onClick={this.closeDeleteConfirmModal}
            >
              <p>No</p>
            </button>
          </div>

        </ReactModal>
      </React.Fragment>
    );
  }
}
BaseMealOptionsCard.defaultProps = {
  connecting: false,
};
BaseMealOptionsCard.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  connecting: PropTypes.bool,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
};
export { BaseMealOptionsCard };
export default connect(state => state)(BaseMealOptionsCard);

