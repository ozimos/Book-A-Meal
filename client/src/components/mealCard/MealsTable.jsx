import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';

import requestServices from '../../services/requestServices';

class MealsTable extends Component {

  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      order: { Meals: [] },
      pages: 1
    };
  }

  onFetchData=(state) => {
    this.setState({ loading: true });
    const { page, pageSize } = state;
    const { id } = this.props;
    const offset = page * pageSize;
    const baseUrl = '/api/v1/orders';
    const url = `${baseUrl}/${id}/meals?limit=${pageSize}&offset=${offset}`;

    requestServices(url)
      .then((res) => {
        this.setState({
          pages: res.data.data.pages,
          order: res.data.data.rows[0],
          loading: false
        });
      });
  }

  render() {

    const columns = [
      {
        headerStyle: { height: '0' },
        id: 'pivot',
        accessor: () => '',
        width: 30
      }, {
        headerStyle: { height: '0' },
        accessor: 'title',
        width: 400
      }, {
        headerStyle: { height: '0' },
        accessor: 'MealOrders.quantity',
        width: 150
      }, {
        headerStyle: { height: '0' },
        accessor: 'price',
        width: 150
      }, {
        headerStyle: { height: '0' },
        aggregate: vals => vals.reduce((accum, val) => accum + val),
        accessor: 'subTotal',
        width: 100
      }
    ];

    const { Meals } = this.state.order;


    return (
      <ReactTable
        ref={(refReactTable) => { this.refReactTable = refReactTable; }}
        data={Meals}
        columns={columns}
        minRows={0}
        defaultPageSize={4}
        pivotBy={['pivot']}
        loading={this.state.loading}
        pages={this.state.pages}
        onFetchData={this.onFetchData}
        sortable={false}
        manual
        collapseOnSortingChange={false}
        collapseOnDataChange={false}
        collapseOnPageChange={false}
        nextText=">>"
        previousText="<<"
      />
    );
  }
}

MealsTable.propTypes = {
  id: PropTypes.string.isRequired,
};
export default MealsTable;
