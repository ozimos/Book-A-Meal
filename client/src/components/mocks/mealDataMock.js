export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
