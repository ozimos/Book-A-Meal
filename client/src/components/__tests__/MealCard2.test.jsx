/* global React:false, shallow:false toJson:false */

import { PlainMealCard2 } from '../mealCard/MealCard2';
import { meal } from '../mocks/mealDataMock';

describe('MealCard2 Component', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<PlainMealCard2
      {...meal.data}
      dispatch={jest.fn()}
      addToMenu={jest.fn()}
      removeFromMenu={jest.fn()}
    />)
      .exists(<div className="card" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<PlainMealCard2
      {...meal.data}
      dispatch={jest.fn()}
      addToMenu={jest.fn()}
      removeFromMenu={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

