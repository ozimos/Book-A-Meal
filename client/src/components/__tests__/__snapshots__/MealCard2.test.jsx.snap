// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCard2 Component renders correctly 1`] = `
<React.Fragment>
  <div
    className="card"
  >
    <img
      alt="Meal"
      className="fluid-img"
      src="nice"
    />
    <div
      className="title-element flexbox wrap"
    >
      <button
        className="modal-open modal-trigger"
        onClick={[Function]}
      >
        nice
      </button>
      <button
        className="btn title-button icon"
        onClick={[Function]}
      >
        ✔
      </button>
      <button
        className="btn title-button icon"
        onClick={[Function]}
      >
        ✖
      </button>
    </div>
  </div>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-content"
    closeTimeoutMS={0}
    contentLabel="Input Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="flexbox"
    >
      <h4
        className="shrink"
      >
        Meal Details
      </h4>
      <div
        className="flexbox"
      >
        <button
          className="btn title-button"
          onClick={[Function]}
        >
          ✖
        </button>
      </div>
    </div>
    <div
      className="menu-info"
    >
      <div>
        nice
      </div>
      <div>
        nice
      </div>
      <div>
        ₦
        200
      </div>
    </div>
  </Modal>
</React.Fragment>
`;
